tests/release_tests/test_launch/utils.py

```diff
@@ -92,3 +110,61 @@ def update_dict(original, updated):
             update_dict(original[key], value)
         else:
             original[key] = value
+
+
+def create_config_files():
+    """Create a launch-config.yml and launch-agent.yml."""
+    launch_config = yaml.load_all(
+        open("wandb/sdk/launch/deploys/kubernetes/launch-config.yaml"),
+        Loader=yaml.Loader,
+    )
+    launch_config_patch = yaml.load_all(
+        open("tests/release_tests/test_launch/launch-config-patch.yaml"),
+        Loader=yaml.Loader,
+    )
+    final_launch_config = []
+    for original, updated in zip(launch_config, launch_config_patch):
+        document = dict(original)
+        update_dict(document, dict(updated))
+        final_launch_config.append(document)
+    yaml.dump_all(
+        final_launch_config,
+        open("tests/release_tests/test_launch/launch-config.yml", "w+"),
+    )
+    launch_agent = yaml.load(
+        open("wandb/sdk/launch/deploys/kubernetes/launch-agent.yaml"),
+        Loader=yaml.Loader,
+    )
+    launch_agent_patch = yaml.load(
+        open("tests/release_tests/test_launch/launch-agent-patch.yaml"),
+        Loader=yaml.Loader,
+    )
+    launch_agent_dict = dict(launch_agent)
+    update_dict(launch_agent_dict, dict(launch_agent_patch))
+    yaml.dump(
+        launch_agent_dict,
+        open("tests/release_tests/test_launch/launch-agent.yml", "w+"),
+    )
+
+
+def init_agent_in_launch_cluster(namespace: str):
+    """Build and deploy the agent in provided cluster namespace."""
+    create_config_files()
+    # Comment out if launch agent image is already built locally
```

Comment:
Can we instead make an image argument for the release tests that run all the tests with a specific image? And only build if not provided?

Also the build step feels like it should be a step outside of each of the test steps and instead in the test setup (in pytests init fixture or in the tox steps)

Reply:
Good idea, I moved the build step out to `pytest_configure` so it's only called once, and only if an image isn't provided.